### ISSUE TITLE
Copy password not working properly after password change

### ISSUE THREAD

**Issue by popxunga** [asbru-cm/asbru-cm#626]
Hi ...
After changing some node password on KeepassXC and saving the password vault, the Copy Password option in the Asbru node context menu keeps providing the old password. 
Asbru should should always get the password from KeepassXC vault I guess.
This is on Loki EL7.
Thanks.

**Comment by gfrenoy** [comment]
Already retrieved passwords are indeed cached, see:

https://github.com/asbru-cm/asbru-cm/blob/loki/lib/PACKeePass.pm#L241-L243

I see several ways to sort this out:

- Remove the cache completely ; it will make things slower.  Don't like this so much.
- Make the cache optional ; so one can choose.  Relatively easy to implement but don't believe it's really user friendly.
- Invalidate cache if a change is detected in the KeePass database.  So when we create the cache, we store the database last modification timestamp ; and before accessing the cache, a check is done to see if the last modification timestamp has changed.  If yes, the cache is emptied and we read the fresh value.

Comments or suggestions are welcome.

**Comment by gfrenoy** [comment]
Just pushed a change that should implement the 3rd option.  Will be ready in a few minutes.

Let me know if it works better for you.

**Comment by popxunga** [comment]
Hi @gfrenoy ...

Thanks a lot for this. I think you took the right approach... the 3rd option is the best one :-)
So I tested last loki build and it works fine.

However I noticed a bunch of messages like this in the terminal which I guess are related with the enhancement:

Use of uninitialized value $timestamp in numeric ne (!=) at /usr/share/asbru-cm/lib/PACKeePass.pm line 501.
Use of uninitialized value in numeric ne (!=) at /usr/share/asbru-cm/lib/PACKeePass.pm line 501.

Looks like a little improvement is needed.
Cheers !
